packages/fx-core/src/plugins/resource/bot/index.ts

```diff
@@ -207,54 +206,19 @@ export class TeamsBot implements Plugin {
       return await runWithExceptionCatching(
         context,
         async () => {
-          if (isCLIDotNetEnabled()) {
-            const node = new QTreeNode({ type: "group" });
-            node.condition = showNotificationTriggerCondition;
-
-            const dotnetNode = new QTreeNode(createHostTypeTriggerQuestionForVS());
-            dotnetNode.condition = {
-              validFunc: async (input: unknown, inputs?: Inputs) => {
-                if (inputs && inputs[CoreQuestionNames.Runtime] === "dotnet") {
-                  return undefined;
-                } else {
-                  return "Runtime is not .net.";
-                }
-              },
-            };
-            node.addChild(dotnetNode);
-
-            const nodejsNode = new QTreeNode(
-              createHostTypeTriggerQuestion(context.answers?.platform)
+          const res = new QTreeNode({
+            type: "group",
+          });
+          const runtimes = [Runtime.Dotnet, Runtime.Node];
+          runtimes.forEach((runtime) => {
+            const node = new QTreeNode(
+              createHostTypeTriggerQuestion(context.answers?.platform, runtime)
             );
-            nodejsNode.condition = {
-              validFunc: async (input: unknown, inputs?: Inputs) => {
-                if (inputs && inputs[CoreQuestionNames.Runtime] === "nodejs") {
-                  return undefined;
-                } else {
-                  return "Runtime is not node.js";
-                }
-              },
-            };
-
-            node.addChild(nodejsNode);
-
-            return ok(node);
-          } else {
-            if (isVSProject(context.projectSettings) || context.answers?.platform === Platform.VS) {
-              const res = new QTreeNode(createHostTypeTriggerQuestionForVS());
-              res.condition = showNotificationTriggerCondition;
-              return ok(res);
-            } else if (isBotNotificationEnabled()) {
-              const res = new QTreeNode({
-                type: "group",
-              });
-              res.addChild(new QTreeNode(createHostTypeTriggerQuestion(context.answers?.platform)));
-              res.condition = showNotificationTriggerCondition;
-              return ok(res);
-            } else {
-              return ok(undefined);
-            }
-          }
+            node.condition = getTriggerQuestionCondition(runtime);
```

Comment:
Nit: The naming seems inconsistent compared to L220. `getTriggerQuestionCondition` vs `showNotificationTriggerCondition`.